### ISSUE TITLE
🤗 [Question]: Issue with Fiber's Static Middleware: Non-Idempotent Behavior When Serving Multiple Vue Projects

### ISSUE THREAD

**Issue by yowaimono** [gofiber/fiber#3180]
### Question Description

# Question Description

I encountered an issue and I'm not sure if it's related to Fiber, as it also occurs with Gin.

I tried to write a small tool using Fiber to deploy my Vue-built `/dist` directory. The code is quite simple:
```go
    app.Static("/uri","/path/dist")
```
Initially, I tested it, and it worked perfectly.
![image](https://github.com/user-attachments/assets/ad18a9d2-13d2-49d6-9428-f9792b28edc0)

I could access my Vue project without any issues. However, after testing it several times, I found that it lacks idempotency; sometimes it doesn't work.

After deploying one Vue project, I switched to another Vue project for deployment, but the page displayed was still from the previous project.

```go
    // app.Static("/uri1","/path/dist1")
    app.Static("/uri2","/path/dist2") // Still accessing dist1
```

However, sometimes it works normally. So I continued my work. I had a requirement to:

```go
    app.Static("/uri1","/path/dist1")
    app.Static("/uri2","/path/dist2") 
```
![image](https://github.com/user-attachments/assets/36df7f04-801c-4228-91cb-01b8475b0e31)

I wanted to start a Fiber server to proxy multiple Vue projects to different URIs and then forward requests on the frontend to achieve multi-Vue project deployment.

However, they either didn't work or only worked for one project, or they might work normally, but not in an idempotent manner.

Initially, I tried to disable caching:

```go
	app.Static("/admin", "./admin/", fiber.Static{
		CacheDuration: 0, // Disable caching
	})
	app.Static("/front", "./front/", fiber.Static{
		CacheDuration: 0, // Disable caching
	})

```
But it still didn't work. 

Although it sometimes works normally, I still want to know the reason.

### Code Snippet (optional)

```go
package main

import (
	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/logger"
)

func main() {
        // My main logic looks like this:
	app := fiber.New()
	app.Use(logger.New())
	app.Static("/admin", "./admin/")
	app.Static("/front", "./front/")

	app.Listen(":4000")
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my questions prior to opening this one.
- [X] I understand that improperly formatted questions may be closed without explanation.

**Comment by JIeJaitt** [comment]
@yowaimono I think it may be a browser cache problem. You can try holding down the Ctrl and ⇧ Shift keys and pressing the R key (or holding down the Ctrl key and pressing the F5 key). On Mac, use the ⌘ Command key instead of the Ctrl key; then Verify your operation again to see the results

**Comment by JIeJaitt** [comment]
@yowaimono Maybe the vue framework will also have its own caching strategy

**Comment by ReneWerner87** [comment]
@yowaimono can you provide a reproducible example?

**Comment by yowaimono** [comment]
@JIeJaitt Thank you for your response. I followed your instructions, and in one of the scenarios I described, even after deploying another project, the previous page was still displayed. I refreshed the browser cache, and it worked correctly. However, in another scenario, as shown in the image below, refreshing the browser did not work.
![image](https://github.com/user-attachments/assets/d036b1c7-4fb8-469d-8836-d23296eeafd4)
Thank you again.
